nx-dev/ui-common/src/lib/headers/header.tsx

```diff
@@ -52,9 +52,38 @@ export function Header({
   // Use the new docs URL when Astro docs are enabled
   const docsUrl = '/docs/getting-started/intro';
   let [isScrolled, setIsScrolled] = useState(false);
-
+  const [starCount, setStarCount] = useState<number | null>(null);
   const router = useRouter();
 
+  const formatStarCount = (count: number): string => {
+  return new Intl.NumberFormat('en-US', {
+    notation: 'compact',
+    maximumFractionDigits: 1
+  }).format(count);
+  };
+
+useEffect(() => {
+  let isMounted = true;
+  const getStarCount = async () => {
+    try {
+      const response = await fetch('https://api.github.com/repos/nrwl/nx');
+      if (!response.ok) {
+        throw new Error(`GitHub API error: ${response.status}`);
+      }
+      const data = await response.json();
+      if (isMounted) {
+        setStarCount(data.stargazers_count);
+      }
+    } catch (error) {
+      console.error('Failed to fetch GitHub stars:', error);
+    }
+  };
+  getStarCount();
+  return () => {
+    isMounted = false;
+  };
+}, []);
```

Comment:
this should not be making external API calls. we already have that info when generating the sites. pass this value into the component to render.